exampleLitActions/src/conditionalSigning.ts

```diff
@@ -0,0 +1,238 @@
+  const value = process.env[name];
+  if (!value) {
+    console.warn(`Environment variable ${name} is not set`);
+  }
+  return value || "";
+};
+
+// Environment variables
+const ETHEREUM_PRIVATE_KEY = getEnv("ETHEREUM_PRIVATE_KEY");
+const CHAIN_TO_CHECK_CONDITION_ON = getEnv("CHAIN_TO_CHECK_CONDITION_ON") || "ethereum";
+const LIT_PKP_PUBLIC_KEY = process.env["LIT_PKP_PUBLIC_KEY"];
+const LIT_CAPACITY_CREDIT_TOKEN_ID = process.env["LIT_CAPACITY_CREDIT_TOKEN_ID"];
+
+// Define the Lit Action code for conditional signing
+const litActionCode = `
+async () =>  {
+  try {
+    // test an access control condition
+    const testResult = await Lit.Actions.checkConditions({
+      conditions,
+      authSig,
+      chain,
+    });
+
+    if (!testResult) {
+      LitActions.setResponse({ response: "address does not have 1 or more Wei on Ethereum Mainnet" });
+      return;
+    }
+
+    const sigShare = await LitActions.signEcdsa({
+      toSign: dataToSign,
+      publicKey,
+      sigName: "sig",
+    });
+  } catch (error) {
+    LitActions.setResponse({ response: error.message });
+  }
+};
+`;
+
+// Result interface for our function
+interface ConditionalSigningResult {
+  success: boolean;
+  signatureResult?: Record<string, any>;
+  response?: string;
+  error?: string;
+}
+
+/**
+ * Execute conditional signing with Lit Protocol
+ * @returns Result of the conditional signing operation
+ */
+export const conditionalSigning = async (): Promise<ConditionalSigningResult> => {
+  let litNodeClient: LitNodeClient;
+  
+  try {
+    // Create ethers wallet for signing
+    const ethersWallet = new ethers.Wallet(
+      ETHEREUM_PRIVATE_KEY,
+      new ethers.providers.JsonRpcProvider(LIT_RPC.CHRONICLE_YELLOWSTONE)
+    );
+
+    console.log("🔄 Connecting to the Lit network...");
+    litNodeClient = new LitNodeClient({
+      litNetwork: LIT_NETWORK.DatilTest,
+      debug: false,
+    });
+    await litNodeClient.connect();
+    console.log("✅ Connected to the Lit network");
+
+    // Initialize Lit Contracts client
+    console.log("🔄 Connecting LitContracts client to network...");
+    const litContracts = new LitContracts({
+      signer: ethersWallet,
+      network: LIT_NETWORK.DatilTest,
+      debug: false,
+    });
+    await litContracts.connect();
+    console.log("✅ Connected LitContracts client to network");
+
+    // Handle PKP (Programmable Key Pair)
+    let pkpPublicKey = LIT_PKP_PUBLIC_KEY;
+    if (!pkpPublicKey) {
+      console.log("🔄 PKP wasn't provided, minting a new one...");
+      const pkpInfo = (await litContracts.pkpNftContractUtils.write.mint()).pkp;
+      pkpPublicKey = pkpInfo.publicKey;
+      console.log("✅ PKP successfully minted");
+      console.log(`ℹ️  PKP token ID: ${pkpInfo.tokenId}`);
+      console.log(`ℹ️  PKP public key: ${pkpPublicKey}`);
+      console.log(`ℹ️  PKP ETH address: ${pkpInfo.ethAddress}`);
+    } else {
+      console.log(`ℹ️  Using provided PKP: ${pkpPublicKey}`);
+    }
```

Comment:
So this is the type of functionality we want to abstract into a util function. It's going to be used by many code examples and isn't directly relevant to the code example which is covering how to conditional signing with a PKP. Really the code to highlight for this code example is in the Lit Action, we just need a PKP public key to send along with the Lit Action request